bottlerocket/agents/src/bin/metal-k8s-cluster-resource-agent/metal_k8s_cluster_provider.rs

```diff
@@ -201,6 +203,27 @@ impl Create for MetalK8sClusterCreator {
                 "Unable to determine the cluster name from the config file ",
             )?
             .to_string();
+        let cluster_version = deserialized_config
```

Comment:
Is this needed for cluster cleanup as well?

Reply:
No, it is not. Only needed for creation.